### PR TITLE
refactor: Separated modules initiator

### PR DESCRIPTION
**Pull request by gazenw**
## Description
Move module's initiator logic to module's package to reduce `cmd_run.go` modification.

## Type of change

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (improvement to existing features and functionality)
- [ ] Documentation update (changes to documentation)
- [ ] Performance improvement (non-breaking change which improves efficiency)
- [x] Code consistency (non-breaking change which improves code reliability and robustness)

## Commit formatting

Please follow the commit message conventions for an easy way to identify the purpose or intention of a commit. Check out our commit message conventions in the [CONTRIBUTING.md](https://github.com/gaze-network/gaze-indexer/blob/main/.github/CONTRIBUTING.md#pull-requests-or-commits)

### CODE REVIEW

File: modules/runes/runes.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	"github.com/gaze-network/indexer-network/pkg/reportingclient"
+	"github.com/gofiber/fiber/v2"
+	"github.com/samber/do/v2"
+	"github.com/samber/lo"
+)
+
+func New(injector do.Injector) (indexer.IndexerWorker, error) {
+	ctx := do.MustInvoke[context.Context](injector)
+	conf := do.MustInvoke[config.Config](injector)
+	btcClient := do.MustInvoke[*rpcclient.Client](injector)
```
Comment (bombnp) [inline modules/runes/runes.go:30]: btcClient should be invoked inside switch case below, as its usage depends on config

File: modules/runes/runes.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	reportingClient := do.MustInvoke[*reportingclient.ReportingClient](injector)
+	httpServer := do.MustInvoke[*fiber.App](injector)
```
Comment (bombnp) [inline modules/runes/runes.go:32]: httpServer should be invoked inside switch case below, as its usage depends on config

File: cmd/cmd_run.go [hunk 3]
```diff
@@ -238,38 +149,61 @@ func runHandler(opts *runCmdOptions, cmd *cobra.Command, _ []string) error {
...
+					if err := indexer.Run(ctx); err != nil {
+						logger.PanicContext(ctx, "Something went wrong, error during running indexer", slogx.Error(err))
+					}
+				}()
+			}
+		}
 	}
 
+	// Run API server
+	httpServer := do.MustInvoke[*fiber.App](injector)
```
Comment (bombnp) [inline cmd/cmd_run.go:196]: There may or may not be an http server, depending on if we have a module with http handlers enabled or not.

Is it possible to start the http server only if such services exist instead?

**Review by bombnp — CHANGES_REQUESTED** [review]

File: cmd/cmd_run.go (hunk 3 above)
Comment (gazenw) [reply cmd/cmd_run.go:196]: It's to complicated to detect and running http server without any enabled module has no effect for now.

**Review by bombnp — APPROVED** [review]

**Pull request merged**